xclim/indices/run_length.py

```diff
@@ -638,6 +638,80 @@ def get_out(rls):
     return da.copy(data=out.transpose(*da.dims).data)
 
 
+def rle_with_holes(
+    da_start: xr.DataArray,
+    window_start: int,
+    index_start: str,
+    da_stop: xr.DataArray | None = None,
+    window_stop: int | None = None,
+    index_stop: str | None = None,
+    dim: str = "time",
+) -> xr.DataArray:
```

Comment:
Not sure if this a good idea to allow `None` here.  Maybe only having `dim` as a kwarg would be better.